### PR TITLE
Feature/git import export

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Add import and export for git repositories via dump file.

### Your checklist for this pull request

- [x] PR is well described and the description can be used as commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by eheimbuch** [comment]
I will update the documentation with the mercurial import/export feature.

File: scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBundleCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    bundleCommand = new GitBundleCommand(mock(GitContext.class));
+    String fileExtension = bundleCommand.getFileExtension();
+    assertThat(fileExtension).isEqualTo("tar");
+  }
+
+  private void addFileToRepoDir(Path repoDir, String filename, String content) throws IOException {
+    Path file = repoDir.resolve(filename);
+    if (!Files.exists(file)) {
+      Files.createFile(file);
+    }
```
Comment (pfeuffer) [inline scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBundleCommandTest.java:76]: Not needed as far as I know

File: scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBundleCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    };
+    return new BundleCommandRequest(byteSink);
+  }
+
+  private void assertStreamContainsContent(ByteArrayOutputStream baos, String content) throws IOException {
+    TarArchiveInputStream tais = new TarArchiveInputStream(new BufferedInputStream(new ByteArrayInputStream(baos.toByteArray())));
+    tais.getNextEntry();
+
+    byte[] result = new byte[(int) tais.getCurrentEntry().getSize()];
+    tais.read(result);
```
Comment (pfeuffer) [inline scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitBundleCommandTest.java:103]:
```suggestion
    byte[] result = IOUtils.toByteArray(tais);
```

File: scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitUnbundleCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    when(gitContext.getDirectory()).thenReturn(temp.toFile());
+    ByteSource byteSource = ByteSource.wrap(baos.toByteArray());
+    UnbundleCommandRequest unbundleCommandRequest = new UnbundleCommandRequest(byteSource);
+
+    unbundleCommand.unbundle(unbundleCommandRequest);
+
+    File createdFile = temp.resolve(filePath).toFile();
+    assertThat(createdFile).exists();
+    assertThat(Files.readLines(createdFile, StandardCharsets.UTF_8).get(0)).isEqualTo(fileContent);
+  }
```
Comment (pfeuffer) [inline scm-plugins/scm-git-plugin/src/test/java/sonia/scm/repository/spi/GitUnbundleCommandTest.java:92]: This test method more or less looks like the first. Some abstractions would be nice to make the differences visible.

File: scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryExportResource.java [hunk 4]
```diff
@@ -126,35 +122,49 @@ public Response exportRepository(@Context UriInfo uriInfo,
...
       .build();
   }
 
-  private String createContentDispositionHeaderValue(Repository repository, boolean compressed) {
+  private String resolveFileExtension(BundleCommandBuilder bundleCommand, boolean compressed) {
+    String fileExtension = bundleCommand.getFileExtension();
+    if (compressed) {
+      fileExtension += ".gz";
+    }
+    return fileExtension;
```
Comment (pfeuffer) [inline scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryExportResource.java:156]: Peronally, I find this more readable:
```suggestion
    if (compressed) {
      return bundleCommand.getFileExtension() + ".gz";
    } else {
      return bundleCommand.getFileExtension();
    }
```

**Review by pfeuffer — CHANGES_REQUESTED** [review]
Some small remarks...

**Pull request merged**